apollo-router/src/executable.rs

```diff
@@ -615,6 +616,17 @@ impl Executable {
             }
         };
 
+        // If there are custom plugins then if RUST_LOG hasn't been set and APOLLO_ROUTER_LOG contains one of the defaults.
+        let user_plugins_present = plugins().filter(|p| !p.is_apollo()).count() > 0;
+        let rust_log_set = std::env::var("RUST_LOG").is_ok();
+        let apollo_router_log = std::env::var("APOLLO_ROUTER_LOG").unwrap_or_default();
+        if user_plugins_present
+            && !rust_log_set
+            && ["", "off", "trace", "debug", "warn", "info"].contains(&apollo_router_log.as_str())
```

Comment:
NIT: I don't think you need to include `""` or `off` since they'll both result in no logging and I imagine the person setting APOLLO_ROUTER_LOG would/should know that.

However, you do need to include `error`.

Reply:
You are right, fixed in [bac15ac](https://github.com/apollographql/router/pull/3540/commits/bac15ac62abf56a764a4b3980a90dac2fee88d72)